common/src/main/java/io/netty/util/Recycler.java

```diff
@@ -496,6 +494,17 @@ boolean scavengeSome() {
         }
 
         void push(DefaultHandle<?> item) {
+            Thread currentThread = Thread.currentThread();
+            if (thread == currentThread) {
+                // The current Thread is the thread that belongs to the Stack, we can try to push the object now.
+                pushNow(item);
+            } else {
+                // The current Thread is not the one that belongs to the Stack, we need to delay the actual push.
+                pushDelayed(item, currentThread);
```

Comment:
`pushDelayed` -> `pushLater`? This naming convention seems to be more of a counterpart for `pushNow`.